### ISSUE TITLE
Use of `~/Library/Application Support` on Mac is user-unfriendly

### ISSUE THREAD

**Issue by bjhargrave** [instructlab/instructlab#1747]
**Describe the bug**

**To Reproduce**
With the changes in #1471, `platformdirs` is used to identify platform specific folders to use. On Mac this results in putting files in `~/Library/Application Support/instructlab`. This includes the models folder, the datasets (formerly generated) folder, and the taxonomy folder. 

On Mac, the `~/Library` folder is hidden in Finder by default. (See <https://www.macworld.com/article/671158/how-to-show-hidden-files-on-a-mac.html>). Which means users won't be able to easily navigate to the `~/Library/Application Support/instructlab` folder in finder to see the files in the their taxonomy or the generated training data.

It is also an awkward file path to type in terminal given the space in the folder name. One must escape or quote.

The support for Mac also does not respect the XDG_ env vars if set. More sophisticated developers will use XDG_ and would expect it to be honored.

`~/Library/Application Support` is for files the user will generally never have to directly touch or see. But there are many files in InstructLab (e.g. taxonomy repo) which we do expect to user to see and touch.

**Expected behavior**

I would expect that the folders are in a place the user can easily access from both Finder and terminal. 

Respect XDG_ env vars if set.

**Screenshots**

**Device Info (please complete the following information):**
 - Hardware Specs: M3 Max, 64GB
```sh
$ ilab system info
sys.version: 3.11.9 | packaged by conda-forge | (main, Apr 19 2024, 18:34:54) [Clang 16.0.6 ]
sys.platform: darwin
os.name: posix
platform.release: 23.5.0
platform.machine: arm64
instructlab.version: 0.17.1.dev494
instructlab-dolomite.version: 0.0.1
instructlab-eval.version: 0.0.9
instructlab-quantize.version: 0.1.0
instructlab-schema.version: 0.2.0
instructlab-sdg.version: 0.1.2
instructlab-training.version: 0.0.5.1
torch.version: 2.3.1
torch.backends.cpu.capability: NO AVX
torch.version.cuda: None
torch.version.hip: None
torch.cuda.available: False
torch.backends.cuda.is_built: False
torch.backends.mps.is_built: True
torch.backends.mps.is_available: True
llama_cpp_python.version: 0.2.79
llama_cpp_python.supports_gpu_offload: True
```
**Additional context**

**Comment by bjhargrave** [comment]
cc: @jjasghar

**Comment by bjhargrave** [comment]
To serve granite-7b, I know have this awkward command to enter.

```sh
ilab model serve --model-path "/Users/hargrave/Library/Application Support/instructlab/models/granite-7b-lab-Q4_K_M.gguf"
```

**Comment by tiran** [comment]
Oh, that's going to impact Linux, too. cc: @gshipley 

Related tickets:
* #1705
* #1309
* #1338

**Comment by cdoern** [comment]
@RobotSail PTAL, this should be a pretty quick fix with the current impl, right?

**Comment by jjasghar** [comment]
I had a problem in a taxonomy qna.yaml and this is what it told me.
```bash
(venv-instructlab-3.11) ➜  instructlab ilab taxonomy diff
knowledge/astronomy/constellations/Phoenix/qna.yaml
ERROR 2024-08-08 14:44:17,980 instructlab.utils:516: Problems found in file /Users/jjasghar/Library/Application Support/instructlab/taxonomy/knowledge/astronomy/constellations/Phoenix/qna.yaml
191:5: [warning] wrong indentation: expected 2 but found 4 (indentation)
Reading taxonomy failed with the following error: 1 total errors found across 1 updated taxonomy files!
(venv-instructlab-3.11) ➜  instructlab
```

Are we expecting someone to copy paste `/Users/jjasghar/Library/Application Support/instructlab/taxonomy/knowledge/astronomy/constellations/Phoenix/qna.yaml` to edit a file? this is not delightful.

**Comment by RobotSail** [comment]
@jjasghar No, we aren't expecting this to be the final experience. We need a clean story around managing and interacting with ilab data such as datasets, models, taxonomy, etc. 

We just need everything stored in one place so that it doesn't matter to the user where they are on the system.

If you would like to try tackling this for the taxonomy, it would be very helpful.

**Comment by jaideepr97** [comment]
in the community call it was suggested that we should try and make sure the default dirs are as consistent with the linux versions as possible, so there is a single set of default directories across both operating systems and people can copy paste paths easily if needed 
so the fix for this issue should try and realign macOS directories to look like the ones used in linux

**Comment by bjhargrave** [comment]
> so the fix for this issue should try and realign macOS directories to look like the ones used in linux

And support XDG_ env vars when set.

**Comment by RobotSail** [comment]
I've re-opened this issue as I don't believe this has been adequately addressed. This is part of a broader issue which I've originally [outlined in the local storage dev-doc](https://github.com/instructlab/dev-docs/pull/104) - which is that we need an easy way to reference local files that are picked up by commands such as `ilab model list`, `ilab data list`, and `ilab eval list` (though this one doesn't exist yet). 

Specifically on the topic of using `~/Library/Application Support` when it comes to manually entering the filepath is certainly user-unfriendly, but this is also the [guideline for storing application data on MacOS](https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW1). However; switching to XDG_ dirs will not fix the problem either, since users are still required to enter the full path when they want to serve a specific model `ilab model serve ~/.local/share/instructlab/models/yada-yada-yada/my-model`. 

If there is interest in MacOS respecting `XDG` directories then that is a separate issue. I've [created an issue](https://github.com/instructlab/instructlab/issues/2244) which would track that implementation in light of  https://github.com/instructlab/instructlab/pull/2202

**Comment by RobotSail** [comment]
Closing this issue as it was resolved in #2202 and we have all agreed to use XDG directories on MacOS as opposed to the standard `~/Library/Application Support` that the guidelines suggest.

Regarding usability, we have the following issues to track components that need to be revamped:
- `ilab model list`: #2200 
- `ilab data list`: #2246 
- ilab eval: We aren't storing any eval results yet, so this is TBD
- ilab taxonomy: TBD - needs an issue created
